crates/context/src/lib.rs

```diff
@@ -748,6 +758,28 @@ impl ContextManager {
         Ok(ids)
     }
 
+    pub fn get_own_signing_key(
+        &self,
+        context_id: &ContextId,
+        own_public_key: &PublicKey,
+    ) -> EyreResult<SigningKey> {
```

Comment:
```suggestion
    ) -> EyreResult<Option<SigningKey>> {
```

we won't always have it, and when we don't, we want to initiate sync